diff --git a/src/w32fns.c b/src/w32fns.c
index 85e9cf8b612..68fdb82dbf5 100644
--- a/src/w32fns.c
+++ b/src/w32fns.c
@@ -145,6 +145,8 @@ static BOOL w32_perform_reconversion (HWND, RECONVERTSTRING*);
 static LRESULT w32_get_ime_documentfeed_string (HWND, RECONVERTSTRING*);
 #endif
 
+static int ime_disabled = 0;
+
 #ifndef IDC_HAND
 #define IDC_HAND MAKEINTRESOURCE(32649)
 #endif
@@ -253,6 +255,8 @@ struct MONITOR_INFO_EX
 DECLARE_HANDLE(HMONITOR);
 #endif
 
+typedef HIMC (WINAPI * ImmAssociateContextEx_Proc) (IN HWND wnd, IN HIMC context, IN DWORD flags);
+
 typedef BOOL (WINAPI * TrackMouseEvent_Proc)
   (IN OUT LPTRACKMOUSEEVENT lpEventTrack);
 typedef LONG (WINAPI * ImmGetCompositionString_Proc)
@@ -305,6 +309,8 @@ typedef struct Emacs_GESTURECONFIG
 typedef BOOL (WINAPI * SetGestureConfig_proc) (HWND, DWORD, UINT,
 					       Emacs_PGESTURECONFIG, UINT);
 
+static ImmAssociateContextEx_Proc associate_contextex_fn = NULL;
+
 static TrackMouseEvent_Proc track_mouse_event_fn = NULL;
 static ImmGetCompositionString_Proc get_composition_string_fn = NULL;
 static ImmGetContext_Proc get_ime_context_fn = NULL;
@@ -5759,6 +5765,11 @@ w32_wnd_proc (HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam)
       goto dflt;
 
     case WM_SETFOCUS:
+      if (ime_disabled)
+        associate_contextex_fn (hwnd, 0, 0);
+      else
+        associate_contextex_fn (hwnd, 0, IACE_DEFAULT);
+
       dpyinfo->faked_key = 0;
       reset_modifiers ();
       if (!w32_kbdhook_active)
@@ -6125,6 +6136,10 @@ w32_wnd_proc (HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam)
 #endif
 #endif /* USE_W32_IME */
 
+    case WM_EMACS_IMM_ASSOCIATE_CONTEXT:
+      associate_contextex_fn (hwnd, 0, lParam);
+      break;
+
     default:
 
 #ifdef USE_W32_IME
@@ -9474,6 +9489,47 @@ conversion_agent_wndproc (HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam)
   return 0;
 }
 
+
+DEFUN ("disable-w32-ime", Fdisable_w32_ime, Sdisable_w32_ime,
+       0, 0, "", doc: /* Disable W32 IME.  */)
+  (void)
+{
+  struct frame *f = SELECTED_FRAME ();
+
+  if (f)
+    {
+      SendMessage (FRAME_W32_WINDOW (f), WM_EMACS_IMM_ASSOCIATE_CONTEXT,
+                   (WPARAM) 0, (LPARAM) 0);
+      ime_disabled = 1;
+    }
+
+  return Qnil;
+}
+
+DEFUN ("enable-w32-ime", Fenable_w32_ime, Senable_w32_ime,
+       0, 0, "", doc: /* Enable W32 IME.  */)
+  (void)
+{
+  struct frame *f = SELECTED_FRAME ();
+
+  if (f)
+    {
+      SendMessage (FRAME_W32_WINDOW (f), WM_EMACS_IMM_ASSOCIATE_CONTEXT,
+                   (WPARAM) 0, (LPARAM) IACE_DEFAULT);
+      ime_disabled = 0;
+    }
+
+  return Qnil;
+}
+DEFUN ("w32-ime-disabled-p", Fw32_ime_disabled_p, Sw32_ime_disabled_p,
+       0, 0, 0, doc: /* Return t if W32 IME is disabled.  */)
+  (void)
+{
+  if (ime_disabled)
+    return Qt;
+  else
+    return Qnil;
+}
 
 /*
   Emacs Lisp function entries
@@ -12623,6 +12679,10 @@ window instead of current window.  */);
 
   /* W32 specific functions */
 
+  defsubr (&Sdisable_w32_ime);
+  defsubr (&Senable_w32_ime);
+  defsubr (&Sw32_ime_disabled_p);
+
   defsubr (&Sw32_define_rgb_color);
   defsubr (&Sw32_display_monitor_attributes_list);
   defsubr (&Sw32_send_sys_command);
@@ -12957,6 +13017,11 @@ void
 globals_of_w32fns (void)
 {
   HMODULE user32_lib = GetModuleHandle ("user32.dll");
+  {
+    HMODULE imm32_lib = GetModuleHandle ("imm32.dll");
+    associate_contextex_fn = (ImmAssociateContextEx_Proc)
+      GetProcAddress (imm32_lib, "ImmAssociateContextEx");
+  }
   /*
     TrackMouseEvent not available in all versions of Windows, so must load
     it dynamically.  Do it once, here, instead of every time it is used.
diff --git a/src/w32term.h b/src/w32term.h
index a602b97c981..74551d3a5f7 100644
--- a/src/w32term.h
+++ b/src/w32term.h
@@ -686,6 +686,8 @@ do { \
 #define UNICODE_NOCHAR 0xFFFF
 #endif
 
+#define WM_EMACS_IMM_ASSOCIATE_CONTEXT (WM_USER+2400)
+
 #define WM_EMACS_START                 (WM_USER + 1)
 #define WM_EMACS_KILL                  (WM_EMACS_START + 0)
 #define WM_EMACS_CREATEWINDOW          (WM_EMACS_START + 1)
