diff --git a/configure.ac b/configure.ac
index ab94a6accc..54d532f707 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2612,30 +2612,32 @@ if test "${HAVE_X11}" = "yes" || test "${HAVE_NS}" = "yes" || test "${HAVE_W32}"
        EMACS_CHECK_MODULES([IMAGEMAGICK], [Wand >= 6.3.5 Wand != 6.8.2])
     fi
 
-    if test $HAVE_IMAGEMAGICK = yes; then
-      OLD_CFLAGS=$CFLAGS
-      OLD_LIBS=$LIBS
-      CFLAGS="$CFLAGS $IMAGEMAGICK_CFLAGS"
-      LIBS="$IMAGEMAGICK_LIBS $LIBS"
-      AC_CHECK_FUNCS([MagickRelinquishMemory MagickExportImagePixels \
-		      MagickMergeImageLayers MagickAutoOrientImage])
-      CFLAGS=$OLD_CFLAGS
-      LIBS=$OLD_LIBS
-      # Check that ImageMagick links.  It does not link on Fedora 25
-      # with './configure CC=clang', as pkg-config outputs flags like
-      # -lomp that work for GCC but not Clang.
-      if test "$ac_cv_func_MagickRelinquishMemory" != yes; then
-	HAVE_IMAGEMAGICK=no
+    if test "${opsys}" != "mingw32"; then
+      if test $HAVE_IMAGEMAGICK = yes; then
+        OLD_CFLAGS=$CFLAGS
+        OLD_LIBS=$LIBS
+        CFLAGS="$CFLAGS $IMAGEMAGICK_CFLAGS"
+        LIBS="$IMAGEMAGICK_LIBS $LIBS"
+        AC_CHECK_FUNCS([MagickRelinquishMemory MagickExportImagePixels \
+		        MagickMergeImageLayers MagickAutoOrientImage])
+        CFLAGS=$OLD_CFLAGS
+        LIBS=$OLD_LIBS
+        # Check that ImageMagick links.  It does not link on Fedora 25
+        # with './configure CC=clang', as pkg-config outputs flags like
+        # -lomp that work for GCC but not Clang.
+        if test "$ac_cv_func_MagickRelinquishMemory" != yes; then
+	  HAVE_IMAGEMAGICK=no
+        fi
       fi
-    fi
-    if test $HAVE_IMAGEMAGICK = yes; then
-      AC_DEFINE([HAVE_IMAGEMAGICK], 1, [Define to 1 if using ImageMagick.])
-    else
-      if test "${with_imagemagick}" != "no"; then
-        AC_MSG_ERROR([ImageMagick wanted, but it does not compile.  Maybe some library files are missing?]);
+      if test $HAVE_IMAGEMAGICK = yes; then
+        AC_DEFINE([HAVE_IMAGEMAGICK], 1, [Define to 1 if using ImageMagick.])
+      else
+        if test "${with_imagemagick}" != "no"; then
+          AC_MSG_ERROR([ImageMagick wanted, but it does not compile.  Maybe some library files are missing?]);
+        fi
+        IMAGEMAGICK_CFLAGS=
+        IMAGEMAGICK_LIBS=
       fi
-      IMAGEMAGICK_CFLAGS=
-      IMAGEMAGICK_LIBS=
     fi
     AC_SUBST([IMAGEMAGICK_CFLAGS])
     AC_SUBST([IMAGEMAGICK_LIBS])
@@ -3876,10 +3878,14 @@ if test "${with_native_compilation}" != "no"; then
     fi
 
     # Check if libgccjit is available.
-    AC_CHECK_LIB(gccjit, gcc_jit_context_acquire, [], [libgccjit_not_found])
+    if test "${opsys}" != "mingw32"; then
+      AC_CHECK_LIB(gccjit, gcc_jit_context_acquire, [], [libgccjit_not_found])
+    fi
     AC_CHECK_HEADERS(libgccjit.h, [], [libgccjit_dev_not_found])
     # Check if libgccjit really works.
-    AC_RUN_IFELSE([libgccjit_smoke_test], [], [libgccjit_broken])
+    if test "${opsys}" != "mingw32"; then
+      AC_RUN_IFELSE([libgccjit_smoke_test], [], [libgccjit_broken])
+    fi
     HAVE_NATIVE_COMP=yes
     case "${opsys}" in
       # mingw32 loads the library dynamically.
